app/controllers/users.rb

```diff
@@ -0,0 +1,37 @@
+get '/sessions/new' do
+  erb :please_sign_in
+end
+
+#when you click sign-in
+post '/sessions' do
+  if params[:sign_in]
+    if login
+      redirect '/'
+    else
+      @message = {:mismatch => "That didn't work, try again."}
+      erb :please_sign_in
+    end
+  elsif params[:register]
+    erb :user_form
+  end
+end
```

Comment:
Why are their three code paths here? Attempting to log in should either pass or fail. Something like:

```ruby
post '/sessions' do
  user = User.find_by(username: params[:username])
  if user.password == params[:password]
  	login_as(user)
    redirect '/'
  else
    @message = "Failed to log in with that username and password."
    erb :"sessions/new"
  end
end
```